python/src/energuide/embedded/window.py

```diff
@@ -0,0 +1,64 @@
+import typing
+from energuide.embedded import area
+from energuide.embedded import code
+from energuide.embedded import insulation
+from energuide.embedded import distance
+from energuide import element
+
+_MILLIMETRES_TO_METRES = 1000
+
+
+class _Window(typing.NamedTuple):
+    label: str
+    window_code: typing.Optional[code.WindowCode]
+    window_insulation: insulation.Insulation
+    width: distance.Distance
+    height: distance.Distance
+
+
+class Window(_Window):
+
+    @classmethod
+    def from_data(cls,
+                  window: element.Element,
+                  window_codes: typing.Dict[str, code.WindowCode]) -> 'Window':
+
+        code_id = window.xpath('Construction/Type/@idref')
+        window_code = window_codes[code_id[0]] if code_id else None
+
+        return Window(
+            label=window.get_text('Label'),
+            window_code=window_code,
+            window_insulation=insulation.Insulation(float(window.xpath('Construction/Type/@rValue')[0])),
+            width=distance.Distance(float(window.xpath('Measurements/@width')[0]) / _MILLIMETRES_TO_METRES),
+            height=distance.Distance(float(window.xpath('Measurements/@height')[0]) / _MILLIMETRES_TO_METRES),
+        )
+
+    @property
+    def _window_area(self) -> area.Area:
+        return area.Area(self.width.metres * self.height.metres)
+
+    def to_dict(self) -> typing.Dict[str, typing.Any]:
+        return {
+            'label': self.label,
+            'rsi': self.window_insulation.rsi,
+            'rvalue': self.window_insulation.r_value,
```

Comment:
What's your opinion on renaming the `rvalue` and `rsi` fields? As they are it's a bit inconsistent with the insulation names from other datatypes.

Reply:
As discussed IRL, I'll push this as is and we'll do another refactor adding "insulation" here and other places later. cc @evadb @sleepycat